SEED-platform/seed
# 2515 refactor/progress bar sits at 33

#### Any background context you want to provide?
When uploading new building files the user will eventually be shown a progress bar modal for "Finding inventory matches". This progress bar has 3 steps: "Geocoding", "Matching data", and "Pairing data". The second step can take a very long time to complete causing users to think that the program has crashed. 

#### What's this PR do?
This PR breaks the "Matching data" step into smaller more granular steps. Upon completion of each granular task, the progress bar will advance. This allows the progress bar to advance more frequently and in smaller steps, preventing the user from believing the program has crashed. 

The number of progress steps has been increased from 3 to 9 
The new steps order is:
- Geocoding
- Matching Data
- Merging Matched Properties (5x)
- Assigning Properties to Views
- Pairing Data

The "Matching Data" step gets hung up on a `.filter(...` function. 
The "Merging Matched Properties" steps are generated while iterating through `matched_id_groups`. Each time 1/5th of the properties have been matched a "Merging Matched Properties" message is generated and the progress bar increases.

#### How should this be manually tested?
- Import a file with 1000+ properties 
- Go through the upload process...
- Click "Confirm mappings & start matching" and watch the progress bar.

#### What are the relevant tickets?
#2515 

#### Screenshots
Example of an import file with 2,500 properties



https://user-images.githubusercontent.com/58446472/142479210-82502043-95aa-49e2-b9e4-3447f2313d61.mov
